pkg/reduce/readloop/readloop.go

```diff
@@ -149,6 +160,10 @@ func (rl *ReadLoop) Process(ctx context.Context, messages []*isb.ReadMessage) {
 
 	// ack successful messages
 	rl.ackMessages(ctx, successfullyWrittenMessages)
+	if len(ctrlMessages) != 0 {
+		rl.ackMessages(ctx, ctrlMessages)
+	}
+
 	// close any windows that need to be closed.
```

Comment:
Can we ack these control messages immediately after reading? because if ```len(successfullyWrittenMessages) == 0```  we return early.

Reply:
or ack them before this if check ```len(successfullyWrittenMessages) == 0```

Reply:
I've fixed it, now the e2e tests should go through fine.